bioio_ome_zarr/writers/ome_zarr_writer_v3.py

```diff
@@ -0,0 +1,541 @@
+        rdefs : Optional[dict]
+            OMERO render settings, under "omero" → "rdefs".
+        creator : Optional[dict]
+            Creator metadata, stored under "_creator".
+
+        Notes
+        -----
+        * Builds "datasets" list, each with a per-level scale transform.
+        * If self.multiscale_scale is provided, inserts it under
+          "coordinateTransformations" of the multiscale entry.
+        * Updates self.root.attrs["ome"] with the assembled block.
+        """
+        # Build per-level datasets list
+        datasets_list: List[dict] = []
+        for lvl in range(self.num_levels):
+            scale_vals: List[float] = []
+            for ax_i in range(self.ndim):
+                if self.axes.types[ax_i] == "space":
+                    scale_vals.append(
+                        self.axes.scales[ax_i] * (self.axes.factors[ax_i] ** lvl)
+                    )
+                else:
+                    scale_vals.append(1.0)
+            datasets_list.append(
+                {
+                    "path": str(lvl),
+                    "coordinateTransformations": [
+                        {"type": "scale", "scale": scale_vals}
+                    ],
+                }
+            )
+
+        # Build the multiscale entry
+        multiscale_entry: Dict[str, Any] = {"name": name or ""}
+
+        # Insert top-level scale transform if provided
+        if self.multiscale_scale is not None:
+            multiscale_entry["coordinateTransformations"] = [
+                {"type": "scale", "scale": self.multiscale_scale}
+            ]
+
+        multiscale_entry["axes"] = axes
+        multiscale_entry["datasets"] = datasets_list
+
+        ome_block: Dict[str, Any] = {
+            "version": "0.5",
+            "multiscales": [multiscale_entry],
+        }
+
+        if channels is not None:
+            omero_block: Dict[str, Any] = {"version": "0.5", "channels": channels}
+            if rdefs is not None:
+                omero_block["rdefs"] = rdefs
+            ome_block["omero"] = omero_block
+
+        if creator:
+            ome_block["_creator"] = creator
+
+        self.root.attrs.update({"ome": ome_block})
+
+    def _suggest_chunks(self, shape: Tuple[int, ...]) -> Tuple[int, ...]:
+        """
+        Suggest chunk shapes aiming for ~64MB per chunk.
+
+        Only "space" axes get larger chunks; non-"space" axes → 1.
+
+        Parameters
+        ----------
+        shape : Tuple[int, ...]
+            The shape of the array at the current level.
+
+        Returns
+        -------
+        Tuple[int, ...]
+            A tuple of chunk sizes for each axis.
+
+        Notes
+        -----
+        * Handles cases where there are exactly 2 or 3 spatial axes.
+        * Allocates roughly sqrt(max elements) along X first, then Y,
+          then Z.
+        """
+        bpe = self.dtype.itemsize
+        maxe = (64 << 20) // bpe  # max elements ≈ 64MB
```

Comment:
why 64?  16MB was originally chosen as an appropriate downloadable chunk size for a web-browser-based viewer.  I suggest making that number a parameter that defaults to 16MB but the caller could specify it.  And actually, I feel like this is sort of a higher level decision than the writer should make.  Ideally an external app like ome-zarr-conversion could provide exact chunk sizes and level shapes to this writer.

Reply:
ok, 16 sounds good

Reply:
ok we now use the old chunk_size_from_memory_target function with some slight upgrades to dimension restrictions. Usage for V2 should remain the same! [4205961](https://github.com/bioio-devs/bioio-ome-zarr/pull/41/commits/42059617d831e6f8c66e7bc708e0c6b6b3676df0)